code/modules/surgery/organs/augments_arms.dm

```diff
@@ -0,0 +1,182 @@
+
+/obj/item/organ/internal/cyberimp/arm/attackby(obj/item/weapon/W, mob/user, params)
+	..()
+	if(istype(W, /obj/item/weapon/screwdriver))
+		if(zone == "r_arm")
+			zone = "l_arm"
+		else
+			zone = "r_arm"
+		slot = zone + "_device"
+		user << "<span class='notice'>You modify [src] to be installed on the [zone == "r_arm" ? "right" : "left"] arm.</span>"
+		update_icon()
+	else if(istype(W, /obj/item/weapon/card/emag))
+		emag_act()
+
+/obj/item/organ/internal/cyberimp/arm/Remove(mob/living/carbon/M, special = 0)
+	Retract()
+	..()
+
+/obj/item/organ/internal/cyberimp/arm/emag_act()
+	return 0
+
+/obj/item/organ/internal/cyberimp/arm/gun/emp_act(severity)
+	if(prob(15/severity))
+		owner << "<span class='warning'>[src] is hit by EMP!</span>"
+		// give the owner an idea about why his implant is glitching
+		Retract()
+	..()
+
+/obj/item/organ/internal/cyberimp/arm/proc/Retract()
+	if(!holder || (holder in src))
+		return
+
+	owner.visible_message("<span class='notice'>[owner] retracts [holder] back into \his [zone == "r_arm" ? "right" : "left"] arm.</span>",
+		"<span class='notice'>[holder] snaps back into your [zone == "r_arm" ? "right" : "left"] arm.</span>",
+		"<span class='italics'>You hear a short mechanical noise.</span>")
+
+	owner.unEquip(holder, 1)
+	holder.loc = src
+	holder = null
+	playsound(get_turf(owner), 'sound/mecha/mechmove03.ogg', 50, 1)
+
+/obj/item/organ/internal/cyberimp/arm/proc/Extend(var/obj/item/item)
+	if(!(item in src))
+		return
+
+	holder = item
+
+	holder.flags |= NODROP
+	holder.unacidable = 1
+	holder.slot_flags = null
+	holder.w_class = 5
+	holder.materials = null
+
+	var/arm_slot = (zone == "r_arm" ? slot_r_hand : slot_l_hand)
+	var/obj/item/arm_item = owner.get_item_by_slot(arm_slot)
+
+	if(arm_item)
+		if(!owner.unEquip(arm_item))
+			owner << "<span class='warning'>Your [arm_item] interferes with [src]!</span>"
+			return
+		else
+			owner << "<span class='notice'>You drop [arm_item] to activate [src]!</span>"
+
+	if(zone == "r_arm" ? !owner.put_in_r_hand(holder) : !owner.put_in_l_hand(holder))
+		owner << "<span class='warning'>Your [src] fails to activate!</span>"
+		return
+
+	// Activate the hand that now holds our item.
+	if(zone == "r_arm" ? owner.hand : !owner.hand)
+		owner.swap_hand()
+
+	owner.visible_message("<span class='notice'>[owner] extends [holder] from \his [zone == "r_arm" ? "right" : "left"] arm.</span>",
+		"<span class='notice'>You extend [holder] from your [zone == "r_arm" ? "right" : "left"] arm.</span>",
+		"<span class='italics'>You hear a short mechanical noise.</span>")
+	playsound(get_turf(owner), 'sound/mecha/mechmove03.ogg', 50, 1)
+
+/obj/item/organ/internal/cyberimp/arm/ui_action_click()
+	if(crit_fail || (!holder && !contents.len))
+		owner << "<span class='warning'>The implant doesn't respond. It seems to be broken...</span>"
+		return
+
+	// You can emag the arm-mounted implant by activating it while holding emag in it's hand.
```

Comment:
It would reqiure changing any item subtype to redirect emag to implant. In any way, the only thing that uses emag actions now is an integrated toolset, and it's far from being ready.